lib/phoenix_component.ex

```diff
@@ -2501,6 +2501,81 @@ defmodule Phoenix.Component do
   This button is optional and only necessary if you want to dynamically add entries.
   You can optionally add a similar button before the `<.inputs_for>`, in the case you want
   to prepend entries.
+
+  > ### A note on accessing a field's `value` {: .warning}
+  >
+  > You may be tempted to access `form[:field].value` or attempt to manipulate
+  > the form metadata in your templates. However, bear in mind that the `form[:field]`
+  > value reflects the most recent changes. For example, an `:integer` field may
+  > either contain integer values, but it may also hold a string, if the form has
+  > been submitted.
+  >
+  > This is particularly noticeable when using `inputs_for`. Accessing the `.value`
+  > of a nested field may either return a struct, a changeset, or raw parameters
+  > sent by the client (when using `drop_param`). This makes the `form[:field].value`
+  > impractical for deriving or computing other properties.
+  >
+  > The correct way to approach this problem is by computing any property either in
+  > your LiveViews, by traversing the relevant changesets and data structures, or by
+  > moving the logic to the `Ecto.Changeset` itself.
+  >
+  > As an example, imagine you are building a time tracking application where:
+  >
+  > - users enter the total work time for a day
+  > - individual activities are tracked as embeds
+  > - the sum of all activities should match the total time
+  > - the form should display the remaining time
+  >
+  > Instead of trying to calculate the remaining time in your template by
+  > doing something like `calculate_remaining(@form)` and accessing
+  > `form[:activities].value`, calculate the remaining time based
+  > on the changeset in your `handle_event` instead:
+  >
+  > ```elixir
+  > def handle_event("validate", %{"tracked_day" => params}, socket) do
+  >   changeset = TrackedDay.changeset(socket.assigns.tracked_day, params)
+  >   remaining = calculate_remaining(changeset)
+  >   {:noreply, assign(socket, :form, to_form(changeset, action: :validate))}
+  > end
+  >
+  > # Helper function to calculate remaining time
+  > defp calculate_remaining(changeset) do
+  >   total = Ecto.Changeset.get_field(changeset)
+  >   activities = Ecto.Changeset.get_embed(changeset, :activities)
+  >   remaining = Enum.reduce(activities, total, fn activity, acc ->
```

Comment:
It might be better to remove `remaining = ` from here as it is the return value of the function anyway and nothing is done with `remaining` there either.

Reply:
Thank you!